openstef/model/regressors/median.py

```diff
@@ -68,6 +122,34 @@ def fit(self, x: pd.DataFrame, y: pd.DataFrame, **kwargs) -> RegressorMixin:
         self.feature_names_ = list(x.columns[x.columns.str.startswith("T-")])
         if len(self.feature_names_) == 0:
             raise ValueError("No lag features found in the input data.")
+        
+        # Check that the lags are evenly spaced, i.e. T-1, T-2, T-3, ...
+        lags_in_min = []
+        for feature in self.feature_names_:
+            if feature.str.endswith("min"):
+                lags_in_min.append(int(feature.split("-")[1].split("min")[0]))
+            elif feature.str.endswith("h"):
+                lags_in_min.append(int(feature.split("-")[1].split("d")[0]) * 60 * 24)
+            else:
+                raise ValueError(
+                    f"Feature name '{feature}' does not follow the expected format."
+                    " Expected format is 'T-<lag_in_minutes>' or 'T-<lag_in_days>d'."
+                )
+        diffs = np.diff(lags_in_min)
+        unique_diffs = np.unique(diffs)
+        if len(unique_diffs) > 1:
+            raise ValueError(
+                "Lag features are not evenly spaced. "
+                f"Got lags with differences: {unique_diffs} min. "
+                "Please ensure that the lag features are generated correctly."
+            )
+        # Check that the lags are in the right order
+        if not all(np.diff(lags_in_min) > 0):
```

Comment:
Can we make a static method that automatically enforeces a certain order? (e.g. alphabetically)